sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/checks/VoidShouldBeUnitCheck.kt

```diff
@@ -59,30 +65,50 @@ class VoidShouldBeUnitCheck : AbstractCheck() {
     // So we check argument types of a typealias separately
     override fun visitTypeAlias(typeAlias: KtTypeAlias, kotlinFileContext: KotlinFileContext) {
 
-        val ktTypeReferences = typeAlias.collectDescendantsOfType<KtTypeReference>()
+        val ktTypeReferences =
+            flattenTypeRefs(typeAlias.getTypeReference()?.typeElement?.typeArgumentsAsTypes ?: return)
 
         kotlinFileContext.bindingContext.get(BindingContext.TYPE_ALIAS, typeAlias)
             ?.underlyingType
             ?.flattenTypeArguments()
             ?.forEachIndexed { i, type ->
                 if (type.isJavaLangVoid()) {
                     kotlinFileContext.reportIssue(
-                        ktTypeReferences[i],
-                        MESSAGE
+                        // Can only be null in case of <*>, but here ktTypeReferences[i] is matched as `Void`
+                        ktTypeReferences[i]!!,
+                        message
                     )
                 }
             }
     }
 }
 
-private fun KotlinType.flattenTypeArguments(): Sequence<KotlinType> =
-     arguments.asSequence().flatMap { it.type.flattenTypeArguments() } + sequenceOf(this)
+private tailrec fun flattenTypeRefs(
+    typeRefs: List<KtTypeReference?>,
+    acc: MutableList<KtTypeReference?> = mutableListOf()
+): List<KtTypeReference?> =
+    if (typeRefs.isEmpty()) acc
+    else flattenTypeRefs(
+        typeRefs = typeRefs.flatMap { it?.typeElement?.typeArgumentsAsTypes ?: emptyList() },
+        acc = acc.apply { addAll(typeRefs) })
+
+private tailrec fun flattenTypeProjections(
+    typeProjections: List<TypeProjection>,
+    acc: MutableList<TypeProjection> = mutableListOf()
+): List<TypeProjection> =
+    if (typeProjections.none()) acc
```

Comment:
I don't necessarily have anything against `.none()`, however, I was a bit confused for a second as to why we don't use `.isEmpty()` here (especially as that is what we use on line 90). Is there a reason not to be consistent here?